corehq/apps/locations/util.py

```diff
@@ -211,6 +214,11 @@ def _write_locations(self, writer, location_type):
         include_consumption = self.include_consumption_flag and location_type.name not in self.administrative_types
         if self.headers_only:
             query = SQLLocation.objects.none()
+        elif self.root_location_id:
```

Comment:
I think this stuff should be abstracted out to a `base_queryset` kind of thing.  There's an `_increment_progress` method that uses the total count of locations on the domain as the total for a progress bar, while this change will of course make that number smaller.

Reply:
292a5968ec3